akka-diagnostics/src/main/resources/reference.conf

```diff
@@ -0,0 +1,56 @@
+##########################################
+# Akka Diagnostics Reference Config File #
+##########################################
+
+
+# This is the reference config file that contains all the default settings.
+# Make your edits/overrides in your application.conf.
+
+# Auto enable the checker
+akka.library-extensions += "akka.diagnostics.DiagnosticsExtension"
+
+#//#starvation-detector
+# The starvation detector is a dedicated thread that schedules a task on the system's
+# main dispatcher and checks if the execution time is less than the threshold. Exceeding the
+# threshold is an indication for thread starvation. The starvation detector will log a warning
+# in that case with statistics and stack traces of the dispatcher's threads.
+akka.diagnostics.starvation-detector {
+  # The interval to check task execution time on the dispatcher.
+  #
+  # Set to 0 or "off" to disable checking.
+  check-interval = 1s
+
+  # The starvation detector only runs after the initial delay. The idea is that during startup,
+  # initial tasks like class loading, warming up JIT and similar environment issues can decrease
+  # the overall throughput. As this is usually a transient condition, checking will only run after
+  # the configured time.
+  #
+  # Set to 0 or "off" to run checking right from the start.
+  initial-delay = 10s
+
+  # The maximum time during which the dispatcher is expected to execute the starvation detection task.
+  # If the dispatcher takes longer to execute the task, the dispatcher (and CPU or other infrastructure
+  # that is needed to run tasks) is considered busy and a warning is logged.
+  #
+  # The default value was chosen to be high enough to decrease the likelihood of false positives while
+  # still being likely to show severe problems early.
+  #
+  # For many applications (and also for internal Akka messaging) much smaller delays can already become
+  # a problem. To detect smaller dispatcher related delays, decrease the value, but keep in mind the
+  # higher chance of false positives.
+  max-delay-warning-threshold = 100 ms
+
+  # The minimum time between consecutive warnings.
+  #
+  # When thread starvation is detected it is likely that it will last for a longer period of time.
+  # This setting can be used to prevent that warnings from the starvation detector flood the logs.
+  warning-interval = 10 seconds
+
+  # When the starvation detector triggers, it will show aggregated information about the state of threads in a pool.
+  # Threads are aggregated by traces and, by default, the top 5 stack traces (by number of threads having the same trace)
+  # are shown.
+  #
+  # Use "infinite" to show aggregated numbers for all threads.
+  thread-traces-limit = 5
+}
+#//#starvation-detector
```

Comment:
add empty line - that actually matters for reference.conf files when they are merged for fat jar building

Reply:
Didn't know that. Good to know :)

Reply:
added